src/components/KTableData/KTableData.vue

```diff
@@ -0,0 +1,576 @@
+<template>
+  <KTableView
+    :cell-attrs="cellAttrs"
+    class="k-table-data"
+    :data="tableData"
+    :empty-state-action-message="emptyStateActionMessage"
+    :empty-state-action-route="emptyStateActionRoute!"
+    :empty-state-button-appearance="getEmptyStateButtonAppearance"
+    :empty-state-icon-variant="emptyStateIconVariant"
+    :empty-state-message="emptyStateMessage"
+    :empty-state-title="emptyStateTitle"
+    :error="error"
+    :error-state-action-message="errorStateActionMessage"
+    :error-state-action-route="errorStateActionRoute!"
+    :error-state-message="errorStateMessage"
+    :error-state-title="errorStateTitle"
+    :headers="tableHeaders"
+    :hide-headers="hideHeaders"
+    :hide-pagination="hidePagination || !showPagination"
+    :hide-pagination-when-optional="hidePaginationWhenOptional"
```

Comment:
It seems `hide-pagination-when-optional` is not implemented in `KTableView`.

Reply:
Fixed now. It's handled slightly differently in KTableView and KTableData since KTableData has more context about data and API responses.